docs/current/guides/723462-use-secrets.md

```diff
@@ -0,0 +1,225 @@
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/use-secrets/host-env/main.go
+```
+
+This code listing assumes the existence of a host environment variable named `GH_SECRET` containing the secret value. It performs the following operations:
+
+- It reads the value of the host environment variable using the `Host().EnvVariable()` method.
+- It creates a Dagger secret with that value using the `Secret()` method.
+- It uses the `WithSecretVariable()` method to return a container with the secret value assigned to an environment variable in the container.
+- The environment variable can then be used in subsequent container operations, as explained previously.
+
+</TabItem>
+<TabItem value="Node.js">
+
+```javascript file=./snippets/use-secrets/host-env/index.mjs
+```
+
+This code listing assumes the existence of a host environment variable named `GH_SECRET` containing the secret value. It performs the following operations:
+
+- It reads the value of the host environment variable using the `host().envVariable()` method.
+- It creates a Dagger secret with that value using the `Secret()` method.
+- It uses the `withSecretVariable()` method to return a container with the secret value assigned to an environment variable in the container.
+- The environment variable can then be used in subsequent container operations, as explained previously.
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/use-secrets/host-env/main.py
+```
+
+This code listing assumes the existence of a host environment variable named `GH_SECRET` containing the secret value. It performs the following operations:
+
+- It reads the value of the host environment variable using the `host().env_variable()` method.
+- It creates a Dagger secret with that value using the `Secret()` method.
+- It uses the `with_secret_variable()` method to return a container with the secret value assigned to an environment variable in the container.
+- The environment variable can then be used in subsequent container operations, as explained previously.
+
+</TabItem>
+</Tabs>
+
+## Use secrets from the host filesystem
+
+Dagger also lets you mount secrets as files within a container's filesystem. This is useful for tools that look for secrets in a specific filesystem location - for example, GPG keyrings, SSH keys or file-based authentication tokens.
+
+As an example, consider the [GitHub CLI](https://cli.github.com/), which reads its authentication token from a file stored in the user's home directory at `~/.config/gh/hosts.yml`. The following code listing demonstrates how to mount this file at a specific location in a container as a secret, so that the GitHub CLI is able to find it.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/use-secrets/host-fs/main.go
```

Comment:
This feels quite clunky. Is there a better way? @dolanor

Reply:
What feels clunky?

Reply:
Do you mean the code example feels clunky?

Reply:
Yes, because we are reading the file into a string, then writing the string back to a file.

Reply:
Yes, I agree.
I hope I can fix `file.secret`, then, it would feel more natural: file -> file.

Reply:
Maybe we should ask @jpadams if he has some users that have some workflow using git in later pipeline stage?